File: Stack/stack.cpp
#include<iostream>
#include <string>
using namespace std;
const int arraySize = 4;
template<class t>
class Stack {
private:
	int top;
	t stack_array[arraySize];
	int counter = 0;
public:

	Stack() {
		top = -1;
	}
	void push(t item) {
		if (top >= arraySize-1) {
			cout << "stack is full" << endl;
		}
		else {
			counter++;
			top++;
			stack_array[top] = item;

		}

	}
	void pop() {
		if (top < 0) cout << "stack is empty" << endl;
		else top--;

	}
	void print_stack() {
		if (top < 0) { cout << "stack is empty"; }
		else{
		cout << '[';
		for (int i = top; i >= 0; i--)
		{
			cout << stack_array[i] << " ";

		}
		cout << "]" << endl;
		}
	}
	void getTop() {
		cout<<stack_array[top] << endl;
	}
	void is_empty() {
		if (top < 0) {
			cout << "True" << endl;
		}
		else {
			cout << "False" << endl;
		}
	}
	int size() {
		return counter;
	}

};
int main() {
	Stack<string> stack_list;
	stack_list.push("ahmad");
	stack_list.push("khaled");
	stack_list.push("zaid");
	stack_list.print_stack();
	cout << stack_list.size() << endl;
}

File: Stack/Stack.py
"""
Stack will follow LIFO rule (Last in first out)
"""

class Stack:
    stack_list = []

    def __init__(self):
        print("Stack created")

    def add(self, number):
        """
        This method is used to add item to the stack
        :param (number) \n
        :return None
        """
        if len(self.stack_list) < 3:
            self.stack_list += [number]
        else:
            print("stack is full")

    def show_stack(self):
        """
        this method is used to print all the stack items
        :return list containing all the stack items
        """
        return self.stack_list[::-1]

    def delete(self):
        """
        This method is used to delete an item from the stack
        :return None
        """
        self.stack_list = self.stack_list[:len(self.stack_list) - 1]

    def is_empty(self):
        """
        this method is used to check if the stack is empty
        :return Boolean value
        """
        return len(self.stack_list) == 0

    def size(self):
        """
        This method is used to know the size of the stack
        :return: integer number referred to the stack size
        """
        return len(self.stack_list)

    def top_stack(self):
        """
        this method is used to get the last element added to the stack
        :return: last element added to the stack
        """
        return self.stack_list[-1] if len(self.stack_list) > 0 else "stack is empty"


new_stack = Stack()
new_stack.show_stack()

print(new_stack.is_empty())
print(new_stack.top_stack())
